visualization/src/svg.cpp

```diff
@@ -0,0 +1,67 @@
+#include <visualization/svg.hpp>
+
+namespace visualization::render {
+
+    SVGDimension::SVGDimension(double height, double width):
+        height(height), width(width) {}
+
+    double SVGDimension::get_width() const {
+        return width;
+    }
+
+    double SVGDimension::get_height() const {
+        return height;
+    }
+
+    SVGRender::SVGRender(const std::string name, const SVGDimension& dimension):
+        name(name), dimension(dimension) {}
+
+    SVGRender::~SVGRender() {}
+
+    void SVGRender::add_rectangle(const geometry::Rectangle& rectangle) {
+        XMLNode node = XMLNode::Builder{}.has_tag("rect")
```

Comment:
Since an XMLNode is invalid without a name, you could do something like: 
```cpp
        XMLNode node = XMLNode::Builder("rect")
            .with_attribute("x", std::to_string(rectangle.get_center().get_x()))
            .with_attribute("y", std::to_string(rectangle.get_center().get_y()))
            .with_attribute("width", std::to_string(rectangle.get_width()))
            .with_attribute("height", std::to_string(rectangle.get_height()))
            .build();
```